dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -276,6 +276,10 @@ private ListenableFuture<Response> handleThrowable(Throwable clientSideThrowable
                 if (requestCanBeRetried() && shouldAttemptToRetry(clientSideThrowable)) {
                     callsiteStacktrace.ifPresent(clientSideThrowable::addSuppressed);
                     Meter retryReason = retryDueToThrowable.apply(clientSideThrowable);
+                    // Retry immediately when we failed to reach the server, since it wasn't a capacity issue
```

Comment:
Just because we are in this code path does not mean we failed to reach the server.